uplinks/freifunk-berlin-uplink-no-tunnel/files/60-ffuplink

```diff
@@ -1,54 +0,0 @@
-#!/bin/sh
-
-. /lib/functions.sh
-. /lib/functions/network.sh
-
-config_load ffwizard
-local sharenet
-config_get sharenet settings sharenet
-
-[ "$sharenet" = 1 ] || exit
-
-if [ "$INTERFACE" = ffuplink ]; then
-  if [ "$ACTION" = ifup ]; then
-    local gateway
-    logger -t ff-userlog "ffuplink connection will be established"
-    network_get_gateway gateway wan
-    ip route add default via ${gateway} dev ffuplink table ffuplink
-    logger -t ff-userlog "ffuplink connection has been established"
-  fi
-  if [ "$ACTION" = ifdown ]; then
-    logger -t ff-userlog "ffuplink connection went down"
-  fi
-fi
-
-[ "$INTERFACE" = wan ] || exit
```

Comment:
Shouldn't we keep the following lines to (de)activate the ffuplink when the wan-interface changes?

Reply:
I don't believe we need it any more.

If I disconnect the cable to WAN, it waits until the DHCP lease times out in either case to loose the IP address.  But when it does, then the ffuplink dev looses it's address at the same time.

If the cable is connected and I run "ifdown wan", then the ffuplink dev immediately goes down.  When I "ifup wan", then the ffuplink dev immediately goes up.

If I change which DHCP network the wan/ffuplink is connected to, then after the lease timeout both devices obtain a new lease on the new network.

The only exception is if the connection is already up, the user changes the connection from one DHCP network to another, then runs "ifup wan"  then wan gets a new address and ffuplink will wait until the lease times out to get a new address.  This use case is also not covered in the hotplug script.

Reply:
It'S all a bit unclear, e.g. do we want to have the uci-interface wan independen from interface ffuplink? 
Will it make sense to have the ffuplink up, when the wan is down?

Anyway, I checked on my box and dropping this script don'T seem to make it more worse ...